### ISSUE TITLE
3.18.0 regression: Invalid code for nullish coalescing with type assertion

### ISSUE THREAD

**Issue by kryops** [alangpierce/sucrase#609]
After the 3.18.0 release, `sucrase` seems to produce invalid code when combining nullish coalescing with a type assertion:

This still works:
```ts
const foo = {
  bar: (baz ?? null) as any
}
```

This stopped working:
```ts
const foo = {
  bar: baz ?? null as any
}
```

https://sucrase.io/#code=const%20foo%20%3D%20%7B%0A%20%20bar%3A%20baz%20%3F%3F%20null%20as%20any%0A%7D

Thanks for looking into it!

**Comment by alangpierce** [comment]
Thanks for reporting! I think I've tracked down the problem and I should be able to release a fix soon.

**Comment by alangpierce** [comment]
The fix is now released as Sucrase 3.18.1. Thanks again, and sorry for the trouble!
